chainer/datasets/image_dataset.py

```diff
@@ -0,0 +1,146 @@
+import os
+
+import numpy
+try:
+    from PIL import Image
+    available = True
+except ImportError as e:
+    available = False
+    _import_error = e
+
+from chainer import dataset
+
+
+class ImageDataset(dataset.Dataset):
+
+    """Dataset of images built from a list of paths to image files.
+
+    This dataset reads an external image file on every call of the
+    :meth:`__getitem__` operator. The path of the image to retrieve is given as
+    a list of strings.
+
+    If a list of labels is also given, then this dataset returns a tuple of an
+    array and a label. Otherwise, it returns a single array.
+
+    Each image is automatically converted to float32 arrays of shape
+    ``channels, height, width``, where ``channels`` represents the number of
+    channels in each pixel (e.g. 1 for grey-scale images, and 3 for RGB-color
+    images).
+
+    **This dataset requires the Pillow package installed.** In order to use
+    this dataset, install Pillow (e.g. by using the command ``pip install
+    Pillow``). Be careful to prepare appropriate libraries for image formats
+    you want to use.
+
+    Args:
+        paths (list of strs): List of paths to image files. The ``i``-th file
+            is used as the ``i``-th example in this dataset. Each path is
+            relative to the root path.
+        labels (array-like): List of labels. If this is None, then the dataset
+            behaves as an unlabeled dataset.
+        root (str): Root directory to retrieve images.
+        dtype: Data type of resulting image arrays.
+
+    """
+    name = 'ImageDataset'
+
+    def __init__(self, paths, labels=None, root='.', dtype=numpy.float32):
+        if not available:
+            raise ImportError('PIL cannot be loaded. Install pillow!\n'
+                              'The actual error:\n' + str(_import_error))
+        self._paths = paths
+        if labels is None:
+            self._labels = None
+        else:
+            if len(paths) != len(labels):
+                raise ValueError('number of paths and labels mismatched')
+            self._labels = numpy.asarray(labels, dtype=numpy.int32)
+        self._root = root
+        self._dtype = dtype
+
+    def __len__(self):
+        return len(self._paths)
+
+    def __getitem__(self, i):
+        path = os.path.join(self._root, self._paths[i])
+        with Image.open(path) as f:
```

Comment:
It may depend on another image library, depending on the format of the actual images it tries to open. Is there an any way to check the appropriate library before trying to open it?